std/algorithm/comparison.d

```diff
@@ -1298,6 +1298,62 @@ MaxType!T max(T...)(T args)
     return cast(typeof(return)) (chooseB ? b : a);
 }
 
+/**
+Iterates the passed range and selects the lowest value matching $(D pred).
+Can use a custom key getter, as defined by $(D keyGetter)
+(which is the identity by default).
+
+Params:
+    r = Range from which the minimum will be selected
+
+Returns: The lowest value according to $(D pred) of the passed-in values.
+*/
+auto selectOne(alias pred, alias keyGetter = (a) => a, Range)(Range r)
```

Comment:
Changed to extremum and removed the key getter.

The idea of the keygetter or `map` is not only syntax sugar, the idea is also to save `mapFun` calls as if I write `extremum!"a.score() > b.score()`"  `score` would be executed twice as often.

I do admit that `map` is usually an inexpensive function or just an access, but here we save in performance and have better usability.

`input.extremum!((a, b) => computeDistanceTo(a) < computeDistanceTo(b))"`
vs.
`input.minElement!computeDistanceTo`